infra/Pos/Communication/PeerState.hs

```diff
@@ -68,11 +73,21 @@ newtype PeerStateHolder m a = PeerStateHolder
                 HasLoggerName, CanLog,
                 MonadFix)
 
+instance m ~ m' => MFunctor' PeerStateHolder m m' where
+    hoist' =
+      -- ahh, deriving by hand.
```

Comment:
1. The problem was that I couldn't make a `MFunctor'` instance without a newtype because `PeerStateCtx` depends on `m`. After I ported `MonadDB`, maybe `MFunctor'` isn't needed anymore, so I'll try again.
2. It's not Rank-1, note the `forall`. GHC doesn't deduce higher-ranked types.